File: docs/vsto/codesnippet/VisualBasic/Trin_VstcoreActionsPaneWordVB/InsertTextControl.vb
Public Class InsertTextControl

    '--------------------------------------------------------------------------
    '<Snippet8>
    Private Sub addText_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles addText.Click

        If Me.getName.Text <> String.Empty Then
            Globals.ThisDocument.showName.Text = Me.getName.Text
        End If

        If Me.getAddress.Text <> String.Empty Then
            Globals.ThisDocument.showAddress.Text = Me.getAddress.Text
        End If

        Me.getName.Text = String.Empty
        Me.getAddress.Text = String.Empty
    End Sub
    '</Snippet8>

End Class

File: docs/vsto/codesnippet/CSharp/Trin_VstcoreActionsPaneWordCS/InsertTextControl.cs
using System;
using System.Windows.Forms;

namespace Trin_VstcoreActionsPaneWordCS
{
    partial class InsertTextControl : UserControl
    {
        //---------------------------------------------------------------------
        //<Snippet9>
        public InsertTextControl()
        {
            InitializeComponent();
            this.addText.Click += new EventHandler(addText_Click);
        }
        //</Snippet9>
        

        //---------------------------------------------------------------------
        //<Snippet8>
        private void addText_Click(object sender, System.EventArgs e)
        {
            if (this.getName.Text != String.Empty) 
            {
                Globals.ThisDocument.showName.Text = this.getName.Text;
            }

            if (this.getAddress.Text != String.Empty)
            {
                Globals.ThisDocument.showAddress.Text = this.getAddress.Text;
            }

            this.getName.Text = String.Empty;
            this.getAddress.Text = String.Empty;
        }
        //</Snippet8>
    }
}
